src/amo/reducers/home.js

```diff
@@ -95,10 +97,14 @@ export const loadHomeAddons = ({
 type Action = FetchHomeAddonsAction | LoadHomeAddonsAction;
 
 const createInternalAddons = (
-  response: ApiAddonsResponse,
+  response: ApiAddonsResponse | null,
 ): Array<AddonType> => {
-  return response.result.results.map((slug) =>
-    createInternalAddon(response.entities.addons[slug]),
+  if (response === null) {
```

Comment:
Even though we have pretty good Flow coverage, I think it's dangerous to allow a `null` `response`. It could lead to subtle bugs. It's also a surprising interface because why would you ask to create internal add-ons out of a null response? I would rather see the reducer fixed at the source:

```diff
diff --git a/src/amo/reducers/home.js b/src/amo/reducers/home.js
index 1f9034ffe..859ee0bd8 100644
--- a/src/amo/reducers/home.js
+++ b/src/amo/reducers/home.js
@@ -139,7 +139,9 @@ const reducer = (
           return null;
         }),
         featuredExtensions: createInternalAddons(featuredExtensions),
-        featuredThemes: createInternalAddons(featuredThemes),
+        featuredThemes: featuredThemes
+          ? createInternalAddons(featuredThemes)
+          : [],
         resultsLoaded: true,
       };
     }

```

This also would make it more clear how that parameter is optional.